driver_test.go

```diff
@@ -338,3 +309,301 @@ func TestIsDdl(t *testing.T) {
 		}
 	}
 }
+
+func TestExecContextDml(t *testing.T) {
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	// Set up test table.
+	_, err = db.ExecContext(ctx,
+		`CREATE TABLE TestExecContextDml (
+			key	INT64,
+			testString	STRING(1024),
+			testBytes	BYTES(1024),
+			testInt	INT64,
+			testFloat	FLOAT64,
+			testBool	BOOL
+		) PRIMARY KEY (key)`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type testExecContextDmlRow struct {
+		key        int
+		testString string
+		testBytes  []byte
+		testInt    int
+		testFloat  float64
+		testBool   bool
+	}
+
+	tests := []struct {
+		name      string
+		given     string
+		when      string
+		then      string
+		wantRows  []testExecContextDmlRow
+		wantError bool
+	}{
+		{
+			name: "insert single tuple",
+			when: `
+				INSERT INTO TestExecContextDml 
+				(key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ) `,
+			then: `SELECT * FROM TestExecContextDml WHERE key = 1`,
```

Comment:
Sorry to be pedantic, but would it be possible to but the `wantRows` / `wantErrors` inside the `then` value? If not that is ok.

Reply:
Wait I may have done it wrong in a different way this time 😅 if when=behaviour and then=expected, shouldn't insert and query both be when, and want(Rows|Err) be then? Also I can do that with a nested struct, but they can't be in the same atomic val without something cursed occurring

Reply:
It is tru that when = behaviour and then = expected, but you are only running the select to check for the expectation, whereas the behaviour you are testing is the insert. So I think your current structure is correct.

On another note, could you provide an example of what you see the nested struct to be like?

Reply:
something like 
```
tests := []struct {
		name      string
		given     string
		when      string
		then 	stuct {
			query		string
			rows		[]testExecContextDmlRow
			wantError	bool
		}
} 
```

Reply:
LGTM

Reply:
Ok I pushed a version with one test changed to nested struct & the rest commented out because looking at it, I think it makes the script messier especially with the while x.y.z  going on at the end. I could just name the fields thenQuery, thenWantRows, thenWantError, might be neater? Or stick with this?

Reply:
It looks good to me, but I will refer to @larkee for the proper go style for this.

Reply:
Changed all to struct version 👍